# EDA on the efap site 

In [1]:
import pandas as pd

In [ ]:
efap = pd.read_csv('/Users/ayemaq/Desktop/CID_Food_Access/data/clean/efap_cleaned.csv')
mapping = pd.read_csv('/Users/ayemaq/Desktop/CID_Food_Access/data/clean/efap_nta_mapping.csv')
df = pd.merge(efap, mapping, on='left')

In [16]:
df.shape

(561, 10)

In [17]:
df['nta_id'].isna().sum()

8

#### insight:
8 out of 561 EFAP sites (1.4%) did not map to an NTA. Because this represents a small proportion of total sites, we proceed by excluding these records from neighborhood-level aggregation.

In [ ]:
df[df["nta_id"].isna()]

efap_id                                   program_name access_type  \
54     81727                    AFRICAN SERVICES COMMITTEE       Pantry   
123    87357                        ST. EDWARD FOOD PANTRY       Pantry   
440    85774                   NEW BEGINNINGS FOOD PANTRY        Pantry   
468    80891                    CORPUS CHRISTI FOOD PANTRY       Pantry   
474    85348  THE LEGACY CENTER COMMUNITY DEVELOPMENT CORP       Pantry   
517    85284                      VETS INC, HOLLIS GARDENS       Pantry   
537    83567                  THE HARDING FORD VISION, INC       Pantry   
550    85710             MUNA SOCIAL SERVICE SOUTH JAMAICA       Pantry   

     has_pantry_access  has_kitchen_access  weekday_available  \
54                   1                   0                  1   
123                  1                   0                  1   
440                  1                   0                  1   
468                  1                   0                  1   
474                  1                   0                  1   
517                  1                   0                  1   
537                  1                   0                  1   
550                  1                   0                  1   

     weekend_available nta_id  lat  lon  
54                   0    NaN  NaN  NaN  
123                  0    NaN  NaN  NaN  
440                  0    NaN  NaN  NaN  
468                  0    NaN  NaN  NaN  
474                  0    NaN  NaN  NaN  
517                  0    NaN  NaN  NaN  
537                  1    NaN  NaN  NaN  
550                  0    NaN  NaN  NaN

In [21]:
df = df.dropna(subset=["nta_id"])
print(df["nta_id"].isna().sum())

0


8 EFAP sites (1.4% of total) lacked geospatial coordinates (latitude and longitude), and therefore could not be mapped to an NTA. Because neighborhood-level aggregation requires valid geographic identifiers, these records were excluded from subsequent analysis. This exclusion represents a minimal proportion of total sites and is unlikely to materially affect results.